bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -725,68 +731,62 @@ public Image(Device device, ImageDataProvider imageDataProvider) {
 	super(device);
 	this.imageDataProvider = imageDataProvider;
 	currentNativeZoom = DPIUtil.getNativeDeviceZoom();
-	ElementAtZoom<ImageData> data =  DPIUtil.validateAndGetImageDataAtZoom(imageDataProvider, getZoom());
-	ImageData resizedData = DPIUtil.autoScaleImageData(device, data.element(), data.zoom());
-	init (resizedData);
+	ElementAtZoom<ImageData> data =  DPIUtil.validateAndGetImageDataAtZoom(imageDataProvider, getCurrentDeviceZoom());
+	ImageData resizedData = DPIUtil.autoScaleImageData(device, data.element(), getCurrentDeviceZoom(), data.zoom());
+	init (resizedData, getCurrentDeviceZoom());
 	init();
 }
 
 /**
  * Update zoom and refresh the Image based on the native zoom level, if required.
  *
- * @param nativeZoom native zoom in % of the monitor on which the image is painted
+ * @param image the image to get the handle of
+ * @param deviceZoomLevel native zoom in % of the monitor on which the image is painted
```

Comment:
its not the native zoom anymore, is it?

Reply:
In the image it is native zoom because we need that for internal_new_GC

Reply:
But the param deviceZoomLevel  is not equal to the nativeZoom, but to DPIUtil.getZoomForAutoscaleProperty(nativeZoom), right?

Reply:
Oh Sorry never mind I thought it was some other method. yes you are right. I'll update the jdoc comment